server/src/test/java/org/jboss/as/server/moduleservice/PathComparatorTestCase.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.server.moduleservice;
+
+import static org.junit.Assert.assertArrayEquals;
+
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Test;
+
+/**
+ * Verifies the oder of path comparator used in ExternalModuleSpecService is the correct.
+ *
+ * @author Yeray Borges
+ */
+public class PathComparatorTestCase {
+
+    @Test
+    public void pathComparatorOrderTest() {
+        String separator = FileSystems.getDefault().getSeparator();
```

Comment:
For this to work on Windows this var is going to need to be some valid initial part of a Windows path, not just the separator.